src/Function/Metric/Structures.agda

```diff
@@ -0,0 +1,93 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some metric structures (not packed up with sets, operations, etc.)
+------------------------------------------------------------------------
+
+-- The contents of this module should usually be accessed via
+-- `Function.Metric`.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary hiding (Symmetric)
+
+module Function.Metric.Structures
+  {a i ℓ₁ ℓ₂ ℓ₃} {A : Set a} {I : Set i}
+  (_≈ₐ_ : Rel A ℓ₁) (_≈ᵢ_ : Rel I ℓ₂) (_≤_ : Rel I ℓ₃) (0# : I) where
+
+open import Algebra.FunctionProperties using (Op₂)
+open import Function.Metric.Core
+open import Function.Metric.Definitions
+open import Level using (_⊔_)
+
+------------------------------------------------------------------------
+-- Proto-metrics
+
+record IsProtoMetric (d : DistanceFunction A I)
+                   : Set (a ⊔ i ⊔ ℓ₁ ⊔ ℓ₂ ⊔ ℓ₃) where
+  field
+    isTotalOrder     : IsTotalOrder _≈ᵢ_ _≤_
```

Comment:
I don't know the aim of this development, but surely this property is too strong if we want the image to be real numbers.

Reply:
Hmm that's an interesting point... To be honest I've only used it over naturals. I've just checked which assumptions I use in my development, and it turns out I only require it to be a preorder. I  I guess it probably would make sense to weaken it therefore. Anybody have any strong opinions on whether we should drop antisymmetry as well?